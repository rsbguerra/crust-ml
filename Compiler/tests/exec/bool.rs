fn print_bool(b: bool) {
    if b { print!("true\n") } else { print!("false\n") }
}

fn main() {
    print_bool(true);
    print_bool(false);
    print_bool(true  && true);
    print_bool(true  && false);
    print_bool(false && true);
    print_bool(false && false);
    print_bool(true  || true);
    print_bool(true  || false);
    print_bool(false || true);
    print_bool(false || false);
    print_bool(!true);
    print_bool(!false);
    if 1+2 == 3 || 1/0 == 0 { print!("yes\n") }
    if 1+2 == 4 && 1/0 == 0 { print!("no\n") }
}
