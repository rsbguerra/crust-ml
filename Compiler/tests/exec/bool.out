true
false
true
false
false
false
true
true
true
false
false
true
yes
